### PR TITLE
[Merged by Bors] - chore: move file iterators to fluvio_storage crate

### PR DESCRIPTION
**Pull request by morenol**

### CODE REVIEW

File: crates/fluvio-storage/Cargo.toml [hunk 1]
```diff
@@ -56,6 +58,7 @@ fluvio-future = { workspace = true, features = ["fixture"] }
 flv-util = { workspace = true, features = ["fixture"] }
 fluvio-socket = { workspace = true,  features = ["file"] }
 fluvio-protocol = { workspace = true,  features = ["fixture"] }
+fluvio-compression = { workspace = true, features = ["compress"] }
```
Comment (sehz) [inline crates/fluvio-storage/Cargo.toml:61]: is this only needed for iterator? if so, make it optional and add to iterators's dependency

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good except a minor nit about feature flag

File: crates/fluvio-storage/Cargo.toml (hunk 1 above)
Comment (morenol) [reply crates/fluvio-storage/Cargo.toml:61]: It is not needed for iterator, It is just used in tests. I added this in `dev-dependencies` only

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**